modules/RTC/RTCUtils.js

```diff
@@ -114,9 +114,14 @@ function getConstraints(um, options) {
         constraints.video = { mandatory: {}, optional: [] };
 
         if (options.cameraDeviceId) {
-            // new style of settings device id (FF only)
-            constraints.video.deviceId = options.cameraDeviceId;
-            // old style
+            // Don't mix new and old style settings for Chromium as this leads
+            // to TypeError in new Chromium versions. @see
+            // https://bugs.chromium.org/p/chromium/issues/detail?id=614716
+            if (!RTCBrowserType.isChrome() && !RTCBrowserType.isOpera()) {
+                // New style of setting device id.
+                constraints.video.deviceId = options.cameraDeviceId;
+            }
+            // Old style.
             constraints.video.optional.push({
```

Comment:
In the 3 cases that you've modified, the if affects the new style only. Why isn't it an if/else that switches between old and new style?